packages/connector-puppeteer/src/connector.ts

```diff
@@ -43,6 +45,7 @@ type ConnectorOptions = {
 
 export default class PuppeteerConnector implements IConnector {
     private _auth: AuthConfig | null;
+    private _base: URL | null = null;
```

Comment:
What do you think about treating this more like `_finalHref` in all of the connectors and storing a `_baseHref` `string` instead of an `URL`? This would help eliminate some of the places where you're using `_base!` or falling back to `_finalHref` (which, if hit, should only occur when `_finalHref === ''` anyway).